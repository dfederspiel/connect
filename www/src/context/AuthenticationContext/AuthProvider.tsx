import React, { useContext, createContext } from 'react';
import { useProvideAuth } from './useProvideAuth';
import { AuthContext } from './types';
import { AuthModule, Mode } from './AuthModule';
/**
 * AuthContext to be passed to any component that depends on user context.
 */

const authContext = createContext<AuthContext>(null as any);
const authModule = new AuthModule(Mode.Client);

// Provider component that wraps your app and makes auth object ...
// ... available to any child component that calls useAuth().
// eslint-disable-next-line react/prop-types
export const AuthProvider = (props: any): JSX.Element => {
  const { origin, children, provider } = props;
  const auth = useProvideAuth(authModule) as AuthContext;
  return <authContext.Provider value={provider || auth}>{children}</authContext.Provider>;
};

// Hook for child components to get the auth object ...
// ... and re-render when it changes.
export const useAuth = (): AuthContext => useContext(authContext);
