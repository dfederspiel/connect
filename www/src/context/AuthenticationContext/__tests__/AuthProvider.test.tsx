import { render } from '@testing-library/react';
import React from 'react';
import { AuthProvider } from '../AuthProvider';

describe('the auth provider', () => {
  it('exists', () => {
    const { asFragment } = render(<AuthProvider></AuthProvider>);
    expect(asFragment()).toMatchSnapshot();
  });
});
