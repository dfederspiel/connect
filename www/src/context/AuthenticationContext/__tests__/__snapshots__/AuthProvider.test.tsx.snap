// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the auth provider exists 1`] = `<DocumentFragment />`;
